### DISCUSSION TITLE
Prevent users from creating/pushing incorrect tags?

### DISCUSSION THREAD

**Discussion by oerp-odoo** [community/community#88554] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I see there is protected tags feature. It reads as "Protected tags can only be created or deleted by users with enhanced permissions defined by your organization owners. ". Though I wonder if I can use this somehow to not allow creating tags that do not match protected tags pattern?

For example developer pushes incorrectly named tag and it will trigger CI etc, when its not supposed to. Or it pushes some local old tag that was deleted on remote.

**Comment by Shahnawazk7944** [comment, marked as answer]
Oh, this is an extension of your previous question. As far as I know, GitHub's Protected Tags feature does not directly prevent the creation of tags that do not match a pattern, but you can try other methods.

1. Define a pattern that encompasses almost all valid tags, leaving very little room for unauthorized creation. For example, v* for release tags, feature/* for feature branches, or hotfix/* for hotfixes.

2. If certain patterns are absolutely forbidden, create separate rules to explicitly block them. For example, block any tag starting with test or wip.

3. Explore third-party tools that offer enhanced tag management features, such as pre-defined patterns or approval workflows. e.g VersionEye, GitProtect.io

**Reply by oerp-odoo** [reply]
Yeah, thats what I did. I created explicit rule to match forbidden (these are old tags that are obsolete and were deleted from remote) tags so only admins can create those.

**Reply by hirasso** [reply]
Can I somehow use this to prevent _any_ creations of tags on some of my branches? I'm planning to distribute my package only from my dist branch and no other branch.
